core/services/gateway/handlers/vault/handler.go

```diff
@@ -269,12 +271,22 @@ func (h *handler) HandleNodeMessage(ctx context.Context, resp *jsonrpc.Response[
 	}
 
 	ar.mu.Lock()
-	ar.responses = append(ar.responses, resp)
+	_, exists := ar.responses[nodeAddr]
+	if exists {
+		l.Errorw("duplicate response from node, ignoring", "nodeAddr", nodeAddr)
+		ar.mu.Unlock()
+		return nil
+	}
+	ar.responses[nodeAddr] = resp
 	ar.mu.Unlock()
 
 	resp, err := h.aggregator.Aggregate(ctx, l, ar, resp)
-	if err != nil {
-		l.Warnw("error aggregating responses", "error", err)
+	switch {
+	case errors.Is(err, errQuorumUnobtainable):
+		l.Error("quorum unobtainable, returning response to user...", "error", err)
```

Comment:
Should this be a critical error?
It almost certainly requires manual intervention from CRE engineers to see why quorum isn't reachable, and liekly almost always a core bug in our implementation?

Also, we should log all responses verbose too with this log, to ease troubleshooting. Cause the logical step would be to check responses and see why they aren't all the same.